### PR TITLE
Meson improvements

### PR DESCRIPTION
**Pull request by chergert**
* Add a libglyphy_dep which can be automatically accessed via subprojects
 * Install a pkgconfig

I do think that, from the GTK standpoint, we'd probably still want to statically link glyphy, but I'll look into that later.

### CODE REVIEW

File: src/meson.build [hunk 1]
```diff
@@ -48,4 +48,18 @@ libglyphy = library('glyphy', glyphy_sources + glyphy_headers + glyphy_shader_so
   version: '0.0.0',
   install: true)
 
+libglyphy_dep = declare_dependency(
+  link_with: libglyphy,
+  include_directories: [confinc],
+  sources: glyphy_sources + glyphy_headers + glyphy_shader_sources)
```
Comment (eli-schwartz) [inline src/meson.build:54]: Why are you adding the original source files from libglyphy here? That will make the other project which links to libglyphy, both link to it and recompile all its sources a second time and include those as primary sources.

File: src/meson.build [hunk 2]
```diff
@@ -48,4 +48,18 @@ libglyphy = library('glyphy', glyphy_sources + glyphy_headers + glyphy_shader_so
...
+
+meson.override_dependency('glyphy', libglyphy_dep)
+
+libglyphy_pc = pkgmod.generate(libglyphy,
+  name: 'glyphy',
+  description: 'High-quality glyph rendering library using OpenGL ES2 shaders',
+  version: meson.project_version(),
+  subdirs: meson.project_name(),
+  install_dir: join_paths(libdir, 'pkgconfig'))
```
Comment (eli-schwartz) [inline src/meson.build:63]: This install_dir is the default so you don't need to override it.

The version defaults to project_version() if you don't specify it.

The name defaults to the same name as the library name of the library specified in the optional first positional argument, which you have. So in this case it should be unneeded.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: src/meson.build (hunk 1 above)
Comment (chergert) [reply src/meson.build:54]: Probably just because I copy pasta'd from a module also doing things wrong.

File: src/meson.build (hunk 2 above)
Comment (chergert) [reply src/meson.build:63]: Cool, will update too

**Pull request merged**